### ISSUE TITLE
Unable to retrieve schema by subject if schema id = 0

### ISSUE THREAD

**Issue by jessicaaustin** [confluentinc/confluent-kafka-python#429]
Description
===========

Unable to retrieve schema by subject if schema id = 0. 

We have a staging and production environment, and in the staging environment the schema in question has id=163 and everything works fine. In production, the id=0 and we can't produce messages.

**Offending line: https://github.com/confluentinc/confluent-kafka-python/blob/master/confluent_kafka/avro/serializer/message_serializer.py#L99-L101 -- should probably change `if not schema_id:` to `if schema_id is None:`**

Partial stacktrace:

```
 File "/opt/conda/lib/python3.6/site-packages/easyavro/producer.py", line 71, in produce
 value=r[1]
 File "/opt/conda/lib/python3.6/site-packages/confluent_kafka/avro/__init__.py", line 64, in produce
 value = self._serializer.encode_record_with_schema(topic, value_schema, value)
 File "/opt/conda/lib/python3.6/site-packages/confluent_kafka/avro/serializer/message_serializer.py", line 101, in encode_record_with_schema
 raise serialize_err(message)
confluent_kafka.avro.serializer.ValueSerializerError: Unable to retrieve schema id for subject obs-with-qc-value
```

Here's the (truncated) schema, from http://SR_PROD_BASE_URL/subjects/obs-with-qc-value/versions/1

```
{
subject: "obs-with-qc-value",
version: 1,
id: 0,
schema: "{ ... }"
}
```

How to reproduce
================

1. Have a schema with id: 0  (*)
1. Try to send a message on a topic with that schema, using `Producer.produce`
1. The producer fails to find the schema: `Unable to retrieve schema id for subject`

\* I'm not sure how exactly we ended up with a schema with id 0 -- I guess it was just the first one created after we set up the schema registry. 

Checklist
=========
Please provide the following information:

 - [X] confluent-kafka-python and librdkafka version (`confluent_kafka.version()` and `confluent_kafka.libversion()`):  
```
>>> confluent_kafka.version()
('0.11.4', 721920)
>>> confluent_kafka.libversion()
('0.11.5', 722431)
```
 - [ ] Apache Kafka broker version:
 - [ ] Client configuration: `{...}`
 - [ ] Operating system:
 - [X] Provide client logs (with `'debug': '..'` as necessary)
 - [ ] Provide broker log excerpts
 - [ ] Critical issue

**Comment by jessicaaustin** [comment]
After some more digging, I think the real bug here is the fact that we have a schema with `id: 0`. 

I tried starting up a new stack (using landoop/fast-data-dev) and creating a schema, and it starts with id 1. I also found your [mock schema registry](https://github.com/confluentinc/confluent-kafka-python/blob/master/tests/avro/mock_schema_registry_client.py#L45) test code and it's pretty clear that the expectation is the first id is 1.
